### ISSUE TITLE
Cloth components are crashing client

### ISSUE THREAD

**Issue by CeebDev** [citizenfx/fivem#3046]
### What happened?

On the latest gamebuild (3407) some components crash the client

### Expected result

It should set the component and not crash

### Reproduction steps

```lua
SetPedCollectionComponentVariation(PlayerPedId(), 6, "mp_m_2024_02", 4, 0, 0);
```

### Importancy

Crash

### Area(s)

FiveM

### Specific version(s)

Gamebuild 3407 / Windows b12168

**Comment by spacevx** [comment]
@Nobelium-cfx

**Comment by prikolium-cfx** [comment]
> @Nobelium-cfx

Thanks, but there is no need to tag developers. I already added this bug to our internal tracking system and developer will be assigned soon.

**Comment by BurnhamR** [comment]
To add to this, We have been having frequent crashes on Gamebuild 3407 when PEDs die with the same stack trace as the cloth components.

**Comment by KillStr3aK** [comment]
not sure if its the same thing, but 3095 (probably lower versions aswell) also been crashing for a while, just one step earlier than this at `sub_141304A3C (0x101)` (autumn-twenty-potato)

the stack goes through the init function of `CStreaming` and crashes somewhere in the game

**Comment by Pundah** [comment]
> not sure if its the same thing, but 3095 (probably lower versions aswell) also been crashing for a while, just one step earlier than this at `sub_141304A3C (0x101)` (autumn-twenty-potato)
> 
> the stack goes through the init function of `CStreaming` and crashes somewhere in the game

I can't say I have the same issue with ``3095``, but ``3258`` is crashing. ``3258`` seems to be stable for me.

**Comment by ElectriziaDev** [comment]
Hello, do you have a deadline for correcting this problem? We based ourselves on the new version to release our new server, saying that it would be corrected before our release, but we are starting to have doubts. Thank you for your feedback. Good day ! :)

**Comment by mickeygeecom** [comment]
> Hello, do you have a deadline for correcting this problem? We based ourselves on the new version to release our new server, saying that it would be corrected before our release, but we are starting to have doubts. Thank you for your feedback. Good day ! :)

+1 on this.

**Comment by Pundah** [comment]
> Hello, do you have a deadline for correcting this problem? We based ourselves on the new version to release our new server, saying that it would be corrected before our release, but we are starting to have doubts. Thank you for your feedback. Good day ! :)

Lol, Maybe in the next 4 years?
I suggest reading [THIS](https://fivem.team/). It is a long read. This exact issue is highlighted.

Exact section: [HERE](https://fivem.team/#currentstateoffivem)
>The Current State of FiveM
>Most servers still use builds 3095 or below, 3095 being the last build made by the original team.

**Comment by TonybynMp4** [comment]
Yeah i wouldn't expect cfx to fix this ever tbh lol

**Comment by prikolium-cfx** [comment]
> ### What happened?
> On the latest gamebuild (3407) some components crash the client
> 
> ### Expected result
> It should set the component and not crash
> 
> ### Reproduction steps
> SetPedCollectionComponentVariation(PlayerPedId(), 6, "mp_m_2024_02", 4, 0, 0);
> ### Importancy
> Crash
> 
> ### Area(s)
> FiveM
> 
> ### Specific version(s)
> Gamebuild 3407 / Windows b12168
> 
> ### Additional information
> _No response_

You can also help the process by providing a crash dump.

**Comment by prikolium-cfx** [comment]
> Yeah i wouldn't expect cfx to fix this ever tbh lol
Could you please keep any github discussions only engineering and issue related.

**Comment by CeebDev** [comment]
> You can also help the process by providing a crash dump.

Is this real ? Why do i need to provide a crash dump when i give you a 100% repro code ? You can't test and get the dump on your side ??

**Comment by KillStr3aK** [comment]
> > You can also help the process by providing a crash dump.
> 
> Is this real ? Why do i need to provide a crash dump when i give you a 100% repro code ? You can't test and get the dump on your side ??

dont expect much from the ctrl:v team

**Comment by prikolium-cfx** [comment]
> > dont expect much from the ctrl:v team
> 
> I think it would be better to try to collaborate and try to prevent this issue from being blocked and forgotten.
> 
> Pattern of the function that crash(b3407): `85 D2 0F 84 ? ? ? ? 48 8B C4 48 89 58 ? 48 89 68 ? 48 89 70 ? 48 89 78 ? 41 56 48 83 EC ? 8D 42` [87fd845e-6f90-45a6-927a-2a94101dfdcd.dmp](https://github.com/user-attachments/files/18870713/87fd845e-6f90-45a6-927a-2a94101dfdcd.dmp)

Thank you Dani.

**Comment by prikolium-cfx** [comment]
> > dont expect much from the ctrl:v team
> 
> I think it would be better to try to collaborate and try to prevent this issue from being blocked and forgotten.
> 
> Pattern of the function that crash(b3407): `85 D2 0F 84 ? ? ? ? 48 8B C4 48 89 58 ? 48 89 68 ? 48 89 70 ? 48 89 78 ? 41 56 48 83 EC ? 8D 42` [87fd845e-6f90-45a6-927a-2a94101dfdcd.dmp](https://github.com/user-attachments/files/18870713/87fd845e-6f90-45a6-927a-2a94101dfdcd.dmp)

Fix is ready, waiting for code review. That's how this component will look after the fix:

![Image](https://github.com/user-attachments/assets/b9e5d3a0-7b77-46cc-829f-97a4689b89f9)

**Comment by CeebDev** [comment]
> Yes, I was just told that some DLC clothes are console exclusive and for PC there are placeholder files that for some strange reason were crashing. Can I see the fix you did, just out of curiosity?

It's weird bcs i check for `IsPedCollectionComponentVariationGen9Exclusive`, also, some of these components are part of outfits and it's returned by the game itself

**Comment by prikolium-cfx** [comment]
> Yes, I was just told that some DLC clothes are console exclusive and for PC there are placeholder files that for some strange reason were crashing. Can I see the fix you did, just out of curiosity?

It was referring non existing shader variable which most likely exist only in Gen9. So I just simply do not set shader variable if it doesn't exist.

**Comment by prikolium-cfx** [comment]
> Yes, I was just told that some DLC clothes are console exclusive and for PC there are placeholder files that for some strange reason were crashing. Can I see the fix you did, just out of curiosity?

That's how this cool component looks in CodeWalker
Exactly same now as on the screenshot from the game

**Comment by prikolium-cfx** [comment]
> Yes, I was just told that some DLC clothes are console exclusive and for PC there are placeholder files that for some strange reason were crashing. Can I see the fix you did, just out of curiosity?

https://github.com/citizenfx/fivem/commit/d48d5b62cec2b71d107eb153338fb4f5415cf788

Here is how it looks in code.

**Comment by DaniGP17** [comment]
> [d48d5b6](https://github.com/citizenfx/fivem/commit/d48d5b62cec2b71d107eb153338fb4f5415cf788)
> 
> Here is how it looks in code.

Thanks

**Comment by whyauthentic** [comment]
Hey, is it already fixed? I'm getting the same issue for all my Players...

**Comment by mickeygeecom** [comment]
> Hey, is it already fixed? I'm getting the same issue for all my Players... 

You just answered your own question.

**Comment by whyauthentic** [comment]
> You just answered your own question.

Yes, but I'm confused by @prikolium-cfx post

**Comment by prikolium-cfx** [comment]
> > You just answered your own question.
> 
> Yes, but I'm confused by [@prikolium-cfx](https://github.com/prikolium-cfx) post

Hi. What exactly confused you? I can explain and add details if you need any

**Comment by whyauthentic** [comment]
> Hi. What exactly confused you? I can explain and add details if you need any

Yes, uhm you closed the issue and somehow wrote something about fix :D Is the problem fixed now or how does it work?

**Comment by prikolium-cfx** [comment]
> > Hi. What exactly confused you? I can explain and add details if you need any
> 
> Yes, uhm you closed the issue and somehow wrote something about fix :D Is the problem fixed now or how does it work?

It's fixed now for clients who use Latest update channel. After some time, when we will see that it's stable and doesn't break anything it will be pushed as a hotfix to beta and production update channels.

**Comment by whyauthentic** [comment]
> It's fixed now for clients who use Latest update channel. After some time, when we will see that it's stable and doesn't break anything it will be pushed as a hotfix to beta and production update channels.

Okay, so i can say my Players to change to Latest and it should work? Is there anyway we can help with stable testing :D

**Comment by prikolium-cfx** [comment]
> > It's fixed now for clients who use Latest update channel. After some time, when we will see that it's stable and doesn't break anything it will be pushed as a hotfix to beta and production update channels.
> 
> Okay, so i can say my Players to change to Latest and it should work? Is there anyway we can help with stable testing :D

You can help for sure, you can ask for example 10 players to test that on Latest and if that's stable for 1-2 hours for them, I will push that to other branches.

**Comment by whyauthentic** [comment]
> > > It's fixed now for clients who use Latest update channel. After some time, when we will see that it's stable and doesn't break anything it will be pushed as a hotfix to beta and production update channels.
> > 
> > 
> > Okay, so i can say my Players to change to Latest and it should work? Is there anyway we can help with stable testing :D
> 
> You can help for sure, you can ask for example 10 players to test that on Latest and if that's stable for 1-2 hours for them, I will push that to other branches.


Okay :) Thank you! Im gonna test it with some of my Players :D

**Comment by kingofkingzcd** [comment]
wild this is still a thing. lol
